lib/reversetunnel/localsite.go

```diff
@@ -523,7 +524,70 @@ func stringOrEmpty(addr net.Addr) string {
 	return addr.String()
 }
 
+func (s *localSite) setupTunnelForOpenSSHEICENode(ctx context.Context, targetServer types.Server) (*awsoidc.OpenTunnelEC2Response, error) {
```

Comment:
I don't know if this method really belongs in `localSite` tbh. Can it be made a part of whatever `s.client` is? That's the client that interacts with AWS API AFAIU.

Reply:
The goal is to call `awsoidc.OpenTunnelEC2` (everything else is required to generate a Token).

`s.client` is only used to fetch the token to use for the AWS API Calls.
Is doesn't actually call any AWS API

I can try to abstract this method under the `awsoidc` package and only require a client that knows how to `GetIntegration` and `GenerateAWSOIDCToken`.